event-app-api/merge_intervals/merge_intervals.js

```diff
@@ -0,0 +1,42 @@
+export function mergeIntervals(testEvents3) {
+  const merged_intervals = [testEvents3[0]];
+  for (let i = 1; i < testEvents3.length; i++) {
+    const currentEvent = testEvents3[i];
+    const lastMergedEvent = merged_intervals[merged_intervals.length - 1];
+
+    let currentEndTime = new Date(currentEvent.endtime);
+    let lastMergeEndtime = new Date(lastMergedEvent.endtime);
+    let currentStartTime = new Date(currentEvent.starttime);
+
+    if (currentStartTime <= lastMergeEndtime) {
+      lastMergedEvent.endtime = Math.max(
+        lastMergeEndtime.getTime(),
+        currentEndTime.getTime()
+      );
+    } else {
+      merged_intervals.push(currentEvent);
+    }
+  }
+
+  return merged_intervals;
+}
+
+export function findGaps(mergedIntervals) {
+  const gaps = [];
+  if (mergedIntervals.length > 1) {
```

Comment:
you don't really need this check. if mergedIntervals was empty the for loop condition wouldn't ever be satisfied so it wouldn't run anyway

Reply:
Okay, I will change that.